br/pkg/storage/s3.go

```diff
@@ -241,7 +244,70 @@ func NewS3Storage( // revive:disable-line:flag-parameter
 	})
 }
 
-func newS3Storage(backend *backuppb.S3, opts *ExternalStorageOptions) (*S3Storage, error) {
+const (
+	ossRAMMetaUrl = "http://100.100.100.200/latest/meta-data/ram/security-credentials/"
+	ossRAMAK      = "AccessKeyId"
+	ossRAMSK      = "AccessKeySecret"
+	ossRAMToken   = "SecurityToken"
+	//
+	domain_aliyun = "aliyuncs.com"
+)
+
+// auto access without ak / sk.
+
+func autoNewCred(qs *backuppb.S3, opts *ExternalStorageOptions) (cred *credentials.Credentials, err error) {
+	if qs.AccessKey != "" && qs.SecretAccessKey != "" {
+		return credentials.NewStaticCredentials(qs.AccessKey, qs.SecretAccessKey, ""), nil
+	}
+	endpoint := qs.Endpoint
+	// if endpoint is empty,return no error and run default(aws) follow.
+	if endpoint == "" {
+		return nil, nil
+	}
```

Comment:
Seems that we can delete this part.

Reply:
delete this part is ok 
but ,maybe other cloud should parse more and more part code ,and it is wasted